spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/AbstractStreamService.java

```diff
@@ -86,14 +83,7 @@ public void deployStream(String name, Map<String, String> deploymentProperties)
 
 	@Override
 	public void undeployStream(String streamName) {
-		StreamDeployment streamDeployment = this.streamDeploymentRepository.findOne(streamName);
-		if (streamDeployment != null) {
-			if (this.streamDeployer != StreamDeployers.valueOf(streamDeployment.getDeployerName())) {
-				throw new IncompatibleStreamDeployerException(streamDeployer.name());
-			}
-			doUndeployStream(streamName);
-			this.streamDeploymentRepository.delete(streamName);
-		}
+		doUndeployStream(streamName);
```

Comment:
There is not point in a separate doUndeployStream method anymore as its calling method (undeployStream) is empty. We can get rid of the doUndeployStream and have the child classes override the undepoyStream instead.